Model/ProductRepository.php

```diff
@@ -1,25 +1,41 @@
 <?php
 namespace SnowIO\AttributeOptionCode\Model;
 
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Phrase;
+use Magento\Store\Model\Store;
 use SnowIO\AttributeOptionCode\Api\Data\ProductInterface;
 use SnowIO\AttributeOptionCode\Api\ProductRepositoryInterface;
 
 class ProductRepository implements ProductRepositoryInterface
 {
     private $vanillaRepository;
     private $productDataMapper;
+    private $storeManager;
 
     public function __construct(
         \Magento\Catalog\Api\ProductRepositoryInterface $vanillaRepository,
-        ProductDataMapper $productDataMapper
+        ProductDataMapper $productDataMapper,
+        \Magento\Store\Model\StoreManagerInterface $storeManager
     ) {
         $this->vanillaRepository = $vanillaRepository;
         $this->productDataMapper = $productDataMapper;
+        $this->storeManager = $storeManager;
     }
 
     public function save(ProductInterface $product, $saveOptions = false)
     {
+        if ($this->storeManager->getStore()->getCode() !== Store::ADMIN_CODE) {
+            if (!$product->getIdBySku($product->getSku())) {
```

Comment:
As far as I know, `getIdBySku()` is not defined on `ProductInterface`, which means this is relying on a specific concretion being used (`Mage\Catalog\Model\Product`) and therefore violating the API. Perhaps you could inject the product resource model instead and call `getIdBySku()` on that directly.

Reply:
![correct](https://i.imgur.com/PpV0evc.png)

Thanks @joshdifabio, rebased.